src/library/scala/util/Using.scala

```diff
@@ -44,20 +44,53 @@ import scala.util.control.{ControlThrowable, NonFatal}
   * import java.io.{BufferedReader, FileReader}
   * import scala.util.{Try, Using}
   *
+  * val files = List("file1.txt", "file2.txt", "file3.txt", "file4.txt")
   * val lines: Try[Seq[String]] = Using.Manager { use =>
-  *   val r1 = use(new BufferedReader(new FileReader("file1.txt")))
-  *   val r2 = use(new BufferedReader(new FileReader("file2.txt")))
-  *   val r3 = use(new BufferedReader(new FileReader("file3.txt")))
-  *   val r4 = use(new BufferedReader(new FileReader("file4.txt")))
+  *   // acquire resources
+  *   def reader(filename: String) = use(new BufferedReader(new FileReader(filename)))
   *
   *   // use your resources here
   *   def lines(reader: BufferedReader): Iterator[String] =
   *     Iterator.continually(reader.readLine()).takeWhile(_ != null)
   *
-  *   (lines(r1) ++ lines(r2) ++ lines(r3) ++ lines(r4)).toList
+  *   files.map(reader).flatMap(lines)
   * }
   * }}}
   *
+  * Composed or "wrapped" resources may be acquired in order of construction,
+  * if "underlying" resources are not closed:
+  * {{{
+  *   def reader(filename: String) = use(new BufferedReader(use(new FileReader(filename))))
+  * }}}
+  *
+  * Custom resources can be registered on construction by requiring an implicit `Manager`.
+  * This ensures they will be released even if composition fails:
+  * {{{
+  * import scala.util.Using
+  *
+  * case class X(x: String)(implicit mgr: Using.Manager) extends AutoCloseable {
+  *   override def close() = println(s"CLOSE $x")
+  *   mgr.acquire(this)
```

Comment:
ctor activity before definitions?
```suggestion
  *   mgr.acquire(this)
  *   override def close() = println(s"CLOSE $x")
```

Reply:
We have no mechanism to say "do this after ctor", there's an issue for `onCreate`; this would be a use case. You don't want to "leak this"; if I accidentally add a LOC after acquire, it blows up, something calls my close and I blow up worse because I'm partially constructed. The closest thing I've wanted to `onCreate` is to use `return` before trailing local definitions as though it's tail position in method.